### PR TITLE
Fix major issues and perms

### PR DESCRIPTION
**Pull request by elenawinters**
- Fixed Character component (you can see other people's equipment and also equip things again)
- Add a /execute command for admins (this still has issues, consider it experimental)
- Coins with 0 value (showing up as a question mark) no longer drop
- Fixes the Teleport GameMessage, as well as the command.
- Fix permissions on /smash
- Fix permissions on /pvp
- Add .bat files to gitignore (I use bat files to build and run the server)

### CODE REVIEW

File: Uchu.World/Handlers/Commands/CharacterCommandHandler.cs [hunk 1]
```diff
@@ -86,6 +87,7 @@ public string ChangeCoin(string[] arguments, Player player)
         {
             if (arguments.Length != 1) return "coin <delta>";
 
+            /// We parse this as an int instead of long, due to the max long causing bugs. Currency maxes out at a long.
```
Comment (MickVermeulen) [inline Uchu.World/Handlers/Commands/CharacterCommandHandler.cs]: Really nitpicky but we generally only use double slash comments inline and use triple slash comments for doc strings

File: Uchu.World/Handlers/Commands/CharacterCommandHandler.cs [hunk 2]
```diff
@@ -777,9 +779,9 @@ public string World(string[] arguments, Player player)
                 return $"Can't find world with ID {id}";
             }
 
-            string WorldName = WorldTable.ZoneName;
+            var path = Path.Combine(UchuServer.Config.ResourcesConfiguration?.GameResourceFolder, Path.Combine("maps", WorldTable.ZoneName.ToLower()));
```
Comment (MickVermeulen) [inline Uchu.World/Handlers/Commands/CharacterCommandHandler.cs:760]: Just wondering, why was this buggy before and how does this fix it? Not that this is wrong or something, I'm just out of the loop.

File: Uchu.World/Handlers/Commands/OperatorCommandHandler.cs [hunk 3]
```diff
@@ -236,5 +237,57 @@ public string Target(string[] arguments, Player player)
 
             return $"Target: {baseCombatAiComponent.Target}";
         }
+
+
+        [CommandHandler(Signature = "execute", Help = "Execute a command as another player", GameMasterLevel = GameMasterLevel.Admin)]
+        public async Task<string> Execute(string[] arguments, Player executor)
+        {
+            List<Player> players = new List<Player>();
+            if (arguments[0] == "*") {
```
Comment (MickVermeulen) [inline Uchu.World/Handlers/Commands/OperatorCommandHandler.cs]: The code in this block doesn't adhere to the overall Uchu / C# style, we generally do:

```
if ()
{
    ... stuff
}
else
{
    ... other stuff
}
```

and

```
foreach ()
{
    ... stuff
}
```

File: Uchu.World/Handlers/Commands/OperatorCommandHandler.cs [hunk 4]
```diff
@@ -236,5 +237,57 @@ public string Target(string[] arguments, Player player)
...
+                if (player == default) return $"No player named {arguments[0]}";
+                players.Add(player);
+            }
+
+            await using var ctx = new UchuContext();
+            var character = await ctx.Characters.Include(c => c.User).FirstAsync(
+                c => c.Id == executor.Id
+            );
+
+            var _com = new List<string>(arguments);
```
Comment (MickVermeulen) [inline Uchu.World/Handlers/Commands/OperatorCommandHandler.cs]: Bit of a odd variable name, if you meant command I would just use `command` here

File: Uchu.World/Handlers/Commands/OperatorCommandHandler.cs [hunk 5]
```diff
@@ -236,5 +237,57 @@ public string Target(string[] arguments, Player player)
...
+
+            var message = string.Join(" ", _com);
+
+            if (!message.StartsWith('/')) {
+                message = "/" + message;
+            }
+
+            if (!SocialHandler.ClientCommands.Contains(message.Split(" ").ElementAt(0)) && message.Split(" ").ElementAt(0) != "/execute" && message.Length > 1) {
+                foreach (Player player in players) {
+                    player.SendChatMessage("You feel a magical power...", PlayerChatChannel.Normal);
```
Comment (MickVermeulen) [inline Uchu.World/Handlers/Commands/OperatorCommandHandler.cs:288]: Lol I like this little extra

File: Uchu.World/Handlers/Commands/OperatorCommandHandler.cs [hunk 6]
```diff
@@ -236,5 +237,57 @@ public string Target(string[] arguments, Player player)
 
             return $"Target: {baseCombatAiComponent.Target}";
         }
+
+
+        [CommandHandler(Signature = "execute", Help = "Execute a command as another player", GameMasterLevel = GameMasterLevel.Admin)]
+        public async Task<string> Execute(string[] arguments, Player executor)
```
Comment (MickVermeulen) [inline Uchu.World/Handlers/Commands/OperatorCommandHandler.cs]: I'd appreciate a doc string here with maybe some example usage

**Review by MickVermeulen — CHANGES_REQUESTED** [review]
Looks good overall, just had some style / documentation requests.

File: Uchu.World/Handlers/Commands/CharacterCommandHandler.cs (hunk 2 above)
Comment (Jettford) [reply Uchu.World/Handlers/Commands/CharacterCommandHandler.cs:760]: Basically the Zone tables are massive messes there are almost no ways to filter whether a world is correct or not but this does filter out most invalid zones

Comment (MickVermeulen) [reply Uchu.World/Handlers/Commands/CharacterCommandHandler.cs:760]: Seems fair.

**Comment by MickVermeulen** [comment]
All looks good, if you tested the TP GM feel free to merge :)

**Review by MickVermeulen — APPROVED** [review]

**Pull request merged**